File: Julia/hentai.jl
using HTTP, FileIO, JSON

URLS = [
    "https://nekobot.xyz/api/image?type=hentai",
    "https://api.waifu.pics/nsfw/waifu",
    "https://api.waifu.pics/nsfw/neko",
    "https://api.waifu.pics/nsfw/blowjob"
]

if !isdir("./Hentai")
    mkdir("./Hentai")
end

i = 0

function main(i)
    while true
        for url in URLS
            res = HTTP.get(url)
            res = JSON.parse(String(res.body))
            println(res)
            if occursin("waifu", url)
                link = res["url"]
            else
                link = res["message"]
            end
            s = split(link, "/")
            Name = pop!(s)
            try
                download(link, "./Hentai/$Name")
                i += 1
                println("Downloaded $i hentai")
            catch e
                @show e
            end
        end
    end
end

main(i)

File: Perl/hentai.pl
use LWP::UserAgent;
use File::Path qw(make_path);
use JSON qw(decode_json);

my @URLS = (
    "https://nekobot.xyz/api/image?type=hentai",
    "https://api.waifu.pics/nsfw/waifu",
    "https://api.waifu.pics/nsfw/neko",
    "https://api.waifu.pics/nsfw/blowjob"
);

if (! -d "./Hentai") {
    make_path("./Hentai");
}

my $i = 0;

sub main {
    my ($i) = @_;
    while (1) {
        foreach my $url (@URLS) {
            my $ua = LWP::UserAgent->new(agent => "Perl script");
            my $res = $ua->get($url);
            print $res->content;
            eval {
                my $json = decode_json($res->content);
                my $link;
                if ($url =~ /waifu/) {
                    $link = $json->{'url'};
                } else {
                    $link = $json->{'message'};
                }
                my @s = split('/', $link);
                my $Name = pop(@s);
                my $ua2 = LWP::UserAgent->new(agent => "Perl script2");
                my $rep = $ua2->mirror($link, "./Hentai/$Name");
                $i++;
                print "Downloaded $i hentai\n";
            };
            if ($@) {
                print "$@\n";
            }
        }
    }
}

main($i);